### PR TITLE
Alerting: miscllaneous UI fixes & improvements

### PR DESCRIPTION
**Pull request by domasx2**
Bunch of small odds & ends:
* Sort out state tag colors. Split `<StateTag />` into `<SilenceStateTag />`, `<AlertStateTag />`, `<AmAlertStateTag  />`, configured with appropriate colors
*  add `fill="outline"` to cancel button where it was missing
* Only show alert manager picker if there's more than 1 option
* Don't allow expanding alert details if there are no details to show
* Don't show "value" field in alert details if value is empty
* Don't show "query" field in rule details for grafana rules. Currently got no way to show read-only queries there that makes sense
* Miscellaneous CSS fixes

Fixes https://github.com/grafana/alerting-squad/issues/130
Fixes https://github.com/grafana/alerting-squad/issues/136
Fixes https://github.com/grafana/alerting-squad/issues/99
Fixes https://github.com/grafana/alerting-squad/issues/105

### CODE REVIEW

File: public/app/features/alerting/unified/components/AlertManagerPicker.tsx [hunk 1]
```diff
@@ -30,20 +33,33 @@ export const AlertManagerPicker: FC<Props> = ({ onChange, current, disabled = fa
...
-      isClearable={false}
-      backspaceRemovesValue={false}
-      onChange={(value) => value.value && onChange(value.value)}
-      options={options}
-      maxMenuHeight={500}
-      noOptionsMessage="No datasources found"
-      value={current}
-      getOptionLabel={(o) => o.label}
-    />
+    <Field className={styles.field} label={disabled ? 'Alert manager' : 'Choose alert manager'} disabled={disabled}>
```
Comment (jackw) [inline public/app/features/alerting/unified/components/AlertManagerPicker.tsx:42]: Is `disabled` needed on both the `Field` and the child `Select`? I thought the Field set the child also? 🤔

File: public/app/features/alerting/unified/components/AlertManagerPicker.tsx [hunk 2]
```diff
@@ -30,20 +33,33 @@ export const AlertManagerPicker: FC<Props> = ({ onChange, current, disabled = fa
...
+      <Select
+        disabled={disabled}
+        width={29}
+        className="ds-picker select-container"
+        isMulti={false}
```
Comment (jackw) [inline public/app/features/alerting/unified/components/AlertManagerPicker.tsx:47]: nit: I believe this defaults to `false`.

File: public/app/features/alerting/unified/components/AlertManagerPicker.tsx [hunk 3]
```diff
@@ -30,20 +33,33 @@ export const AlertManagerPicker: FC<Props> = ({ onChange, current, disabled = fa
...
+        isClearable={false}
```
Comment (jackw) [inline public/app/features/alerting/unified/components/AlertManagerPicker.tsx:48]: nit: I believe this defaults to `false`.

**Review by jackw — APPROVED** [review]
You might want to wait for someone with better domain knowledge before merging as I'm not very familiar with the project but the code looks good to me! 👍

**Comment by domasx2** [comment]
> You might want to wait for someone with better domain knowledge before merging as I'm not very familiar with the project but the code looks good to me!

Thanks!!!

**Review by nathanrodman — APPROVED** [review]
Nice changes!

File: public/app/features/alerting/unified/components/AlertManagerPicker.tsx (hunk 1 above)
Comment (domasx2) [reply public/app/features/alerting/unified/components/AlertManagerPicker.tsx:42]: yup, confusingly, it does. Removed this prop from the `<Select />`

File: public/app/features/alerting/unified/components/AlertManagerPicker.tsx (hunk 2 above)
Comment (domasx2) [reply public/app/features/alerting/unified/components/AlertManagerPicker.tsx]: removed

File: public/app/features/alerting/unified/components/AlertManagerPicker.tsx (hunk 3 above)
Comment (domasx2) [reply public/app/features/alerting/unified/components/AlertManagerPicker.tsx]: removed

**Pull request merged**